src/compiler/types.ts

```diff
@@ -1099,6 +1118,13 @@ namespace ts {
         _statementBrand: any;
     }
 
+    // Represents a statement that is elided as part of a transformation to emit comments on a
+    // not-emitted node.
+    // @internal
+    // @kind(SyntaxKind.NotEmittedStatement)
+    export interface NotEmittedStatement extends Statement {

```

Comment:
I guess you don't need it now, but I wonder if you could just have a `NotEmittedExpression` and let `StatementExpression` do the heavy lifting.